web/frontend/src/api/navigation.ts

```diff
@@ -33,44 +38,55 @@ export const setWaypoint = (client: Client, lat: number, lng: number, name: stri
   rcLogConditionally(request);
 
   return new Promise((resolve, reject) => {
-    client.navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) =>
+    get(client).navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) =>
       (error ? reject(error) : resolve(response)));
   });
 };
 
-export const getWaypoints = (client: Client, name: string) => {
+const formatWaypoints = (list: navigationApi.Waypoint[]) => {
+  return list.map((item) => {
+    const location = item.getLocation();
+    return {
+      id: item.getId(),
+      lng: location?.getLongitude() ?? 0,
+      lat: location?.getLatitude() ?? 0,
+    };
+  });
+};
+
+export const getWaypoints = (name: string) => {
   const req = new navigationApi.GetWaypointsRequest();
   req.setName(name);
 
   rcLogConditionally(req);
 
-  return new Promise<navigationApi.Waypoint[]>((resolve, reject) => {
-    client.navigationService.getWaypoints(req, new grpc.Metadata(), (error, response) =>
-      (error ? reject(error) : resolve(response?.getWaypointsList() ?? [])));
+  return new Promise<Waypoint[]>((resolve, reject) => {
+    get(client).navigationService.getWaypoints(req, new grpc.Metadata(), (error, response) =>
+      (error ? reject(error) : resolve(formatWaypoints(response?.getWaypointsList() ?? []))));
```

Comment:
Can the work of `formatWaypoint` be moved outside of the `new Promise` constructor?

Reply:
No, unfortunately since it's formatting the response object :(

Reply:
Something like this will work, no? Advantage is that if `formatWaypoints` or `getWaypointsList` throws synchronously, the promise will reject instead of hanging forever (with an uncaught exception)

```ts
return new Promise<Waypoint[]>((resolve, reject) => {
    get(client).navigationService.getWaypoints(req, new grpc.Metadata(), (error, response) =>
      (error ? reject(error) : resolve(response)));
  }).then((response) => formatWaypoints(response?.getWaypointsList() ?? []));
```

Reply:
Ah! Had not considered that. Updated!